ckan/lib/helpers.py

```diff
@@ -1126,8 +1127,53 @@ def _make_menu_item(menu_item, title, **kw):
 
 
 @core_helper
-def default_group_type():
-    return str(config.get('ckan.default.group_type', 'group'))
+def default_group_type(type_='group'):
+    """Get default group/organization type for using site-wide.
+    """
+    return str(config.get('ckan.default.{}_type'.format(type_), type_))
+
+
+@core_helper
+def humanize_entity_type(entity_type, object_type, purpose):
+    """Convert machine-readable representation of package/group type into
+    human-readable form.
+    Example::
+
+      >>> humanize_entity_type('group', 'custom_group', 'add link')
+      'Add Custom Group'
+      >>> humanize_entity_type('group', 'custom_group', 'breadcrumb')
+      'Custom Groups'
+
+    """
+    if entity_type == object_type:
+        return  # use the default text included in template
+
+    log.debug(
+        u'Humanize %s of type %s for %s', entity_type, object_type, purpose)
+    templates = {
+        u'add link': _(u"Add {object_type}"),
+        u'breadcrumb': _(u"{object_type}s"),
+        u'facet label': _(u"{object_type}s"),
+        u'page title': _(u"{object_type}s"),
+        u'main nav': _(u"{object_type}s"),
+        u'create title': _(u"Create {object_type}"),
+        u'content tab': _(u"{object_type}s"),
+        u'no label': _(u"No {object_type}"),
+        u'form label': _(u"{object_type} Form"),
+        u'edit label': _(u"Edit {object_type}"),
+        u'update label': _(u"Update {object_type}"),
+        u'create label': _(u"Create {object_type}"),
+        u'save label': _(u"Save {object_type}"),
+        u'my label': _(u"My {object_type}s"),
+        u'no any objects': _(
+            u"There are currently no {object_type}s for this site"),
+        u'description placeholder': _(
+            u"A little information about my {object_type}..."),
+    }
+    if purpose not in templates:
+        return
+    type_label = object_type.replace(u"_", u" ").capitalize()
```

Comment:
```suggestion
    type_label = object_type.replace(u"_", u" ").capitalize()
    if purpose not in templates:
        return type_label
```

Reply:
We've discussed it in [previous thread](https://github.com/ckan/ckan/pull/5337#discussion_r451040241) - it won't allow using default message from template. And in that way we have to put into `templates` dictionary every single message, even if it very long and unlikely to be changed, like [this one](https://github.com/ckan/ckan/pull/5337/files#diff-ea325fbc1a770c95b6cffacc7f23ecb7R39)

Reply:
@smotornyuk I think this is required because you removed a couple keys from the dict above, and it seems like a sensible default for unknown purpose values

Reply:
ok then, let's go with it

Reply:
I've updated docstring a bit, mentioning default value